### ISSUE TITLE
"gauge --failed" runs all specs from "specs/" subdirectory when `FailedItems` is empty in .gauge/failed.json

### ISSUE THREAD

**Issue by kaustavdm** [getgauge/gauge#418]
### Expected behavior

`gauge --failed` should attempt to run only failed scenarios from previous run. If `FailedItems` is empty in `.gauge/failed.json`, it should not run scenarios or specifications.

### Actual behavior

`gauge --failed` currently runs all the specs in the `specs/` subdirectory in the project root even when `FailedItems` is empty. That `specs/` directory seems to be hard-coded because `gauge --failed` does not accept paths to specs. So, you get the same behaviour on running `gauge --failed <path-to-other-specs-dir>`.

### Steps to reproduce
1. Run `gauge --failed` on a test suite with no previous failures.

### Gauge version

Run gauge -v on your system and paste the results here.

```
Gauge version: 0.4.1.nightly-2016-05-27

Plugins
-------
html-report (2.1.1)
java (0.4.1)
js (1.2.0)
ruby (0.2.0)
xml-report (0.1.2)
```

**Comment by sguptatw** [comment]
gauge --failed . from a subfolder is giving the following error.
```
Failed to read last run information. Reason: File .gauge/failures.json doesn't exist.
2016/06/07 13:38:21 Failed to read last run information. Reason: File .gauge/failures.json doesn't exist.
```

**Comment by mahendrakariya** [comment]
Fix should be available in nightly >= 2016-06-08.

**Comment by sguptatw** [comment]
When there are multiple tests failing, repeatedly running gauge --failed is doubling the statistics everytime.

```
Successfully generated html-report to => /Users/sswaroop/work/gauge-test-apps/basic-java/reports/html-report
Specifications:	1 executed	0 passed	1 failed	0 skipped
Scenarios:	32 executed	0 passed	32 failed	0 skipped
```

```
Successfully generated html-report to => /Users/sswaroop/work/gauge-test-apps/basic-java/reports/html-report
Specifications:	1 executed	0 passed	1 failed	0 skipped
Scenarios:	64 executed	0 passed	64 failed	0 skipped
```

**Comment by apoorvam** [comment]
Fix should be available in nightly >= 10.06.2016

**Comment by sguptatw** [comment]
Html report is showing wrong results when the spec is executed with data driven execution.

Atleast it should show the row with the failure.

**Comment by apoorvam** [comment]
Fix should be available in nightly >= 13.06.2016
